pkg/test/framework/api/components/apps.go

```diff
@@ -43,35 +43,30 @@ type Apps interface {
 	GetAppOrFail(name string, t testing.TB) App
 }
 
+// Service represents a deployed service within the mesh.
+type Service interface {
+	ClusterIP() string
+}
+
 // App represents a deployed fake App within the mesh.
 type App interface {
 	Name() string
 	Endpoints() []AppEndpoint
 	EndpointsForProtocol(protocol model.Protocol) []AppEndpoint
 	Call(e AppEndpoint, opts AppCallOptions) ([]*echo.ParsedResponse, error)
 	CallOrFail(e AppEndpoint, opts AppCallOptions, t testing.TB) []*echo.ParsedResponse
+	Service() Service
 }
 
 // AppCallOptions defines options for calling a DeployedAppEndpoint.
 type AppCallOptions struct {
-	// Secure indicates whether a secure connection should be established to the endpoint.
-	Secure bool
```

Comment:
why are these removed?

Reply:
There are 2 reasons for this:
First, this is not really necessary, because it's already part of the ```AppEndpoint```. If you try to get an ```AppEndpoint``` from an application using ```EndpointsForProtocol(model.Protocol)```, you already specify the type of protocol, you would like to use. Therefore, it's not necessary to pass this as option to ```Call(AppEndpoint,...)```
Second, it was not used.

Reply:
I think the goal here was to allow the client to make incorrect calls to the target.  So keeping these things in CallOptions was the only way to afford that ability in the API.

Also, headers are generally needed by most tests.